ratpack-cassandra/src/main/java/smartthings/ratpack/cassandra/CassandraSessionFactory.java

```diff
@@ -0,0 +1,24 @@
+package smartthings.ratpack.cassandra;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+public final class CassandraSessionFactory implements Provider<Session> {
+
+	private final Cluster cluster;
+	private final CassandraModule.Config config;
+
+	@Inject
+	CassandraSessionFactory(Cluster cluster, CassandraModule.Config config) {
+		this.cluster = cluster;
+		this.config = config;
+	}
+
+	@Override public Session get() {
+		return (config.keyspace != null && !"".equals(config.keyspace)) ?
+				cluster.connect(config.keyspace) :
+				cluster.connect();
```

Comment:
This was previously done in the Service onStart method.  Moving it here makes it simple to change to a different Session implementation for tracing.